evdev/uinput.py

```diff
@@ -281,11 +283,75 @@ def _verify(self):
             msg = 'uinput device name must not be longer than {} characters'
             raise UInputError(msg.format(_uinput.maxnamelen))
 
-    def _find_device(self):
+    def _find_device(self, fd):
+        '''
+        Tries to find the device node. Will delegate this task to one of
+        several platform-specific functions.
+        '''
+        if platform.system() == 'Linux':
+            try:
+                sysname = _uinput.get_sysname(fd)
+                return self._find_device_linux(sysname)
+            except OSError:
+                # UI_GET_SYSNAME returned an error code. We're likely dealing with
+                # an old kernel. Guess the device based on the filesystem.
+                pass
+
+        # If we're not running or Linux or the above method fails for any reason,
+        # use the generic fallback method.
+        return self._find_device_fallback()
+
+    def _find_device_linux(self, sysname):
+        '''
+        Tries to find the device node when running on Linux.
+        '''
+
+        syspath = f'/sys/devices/virtual/input/{sysname}'
+
+        # The sysfs entry for event devices should contain exactly one folder
+        # whose name matches the format "event[0-9]+". It is then assumed that
+        # the device node in /dev/input uses the same name.
+        regex = re.compile('event[0-9]+')
+        for entry in os.listdir(syspath):
+            if regex.fullmatch(entry):
+                device_path = f'/dev/input/{entry}'
+                break
+        else:  # no break
+            raise FileNotFoundError()
+
+        # It is possible that there is some delay before /dev/input/event* shows
+        # up on old systems that do not use devtmpfs, so if the device cannot be
+        # found, wait for a short amount and then try again once.
+        try:
+            return device.InputDevice(device_path)
+        except FileNotFoundError:
+            time.sleep(0.1)
```

Comment:
If the file shows up late, then the loop above will throw `raise FileNotFoundError()` in line 320, so it won't ever reach the time.sleep

Reply:
If the file shows up late, would that affect both `/sys/devices/virtual/input/foo/event5` and `/dev/input/event5`? If yes, then this line will never be reached

Reply:
Whether `/sys/devices/virtual/input/foo` shows up depends on whether sysfs is mounted at `/sys`, and whether `/dev/input/event*` shows up late depends on whether devtmpfs is mounted as `/dev`, so these two are somewhat independent from each other.

The reason I added the try-except to checking `/dev` but not to `/sys` is because it is possible to have a functional `/dev` system without using devtmpfs. The "special" files that you find in `/dev` are actually just "device nodes", and those device nodes can also be created using `mknod`.

However, I am not aware of any alternative to sysfs for `/sys`. Sysfs exposes kernel data structures to userspace, and those data structures do not have [device node numbers](https://www.kernel.org/doc/Documentation/admin-guide/devices.txt) that you can just `mknod` into existence.

Based on that, I assume that `/sys` shows up either instantaneously or never (for crazy operating systems that do not mount `/sys`), but `/dev` might show up with delay.